pyteal.ini

```diff
@@ -0,0 +1,5 @@
+[pyteal]
+
+[pyteal-source-mapper]
+enabled = False
+debug = False
```

Comment:
I'm not really convinced of this file, still, after messing around with it as a user for a few days.  Not really convinced that it is necessary to touch the filesystem like this, and this is not the feature that makes me think there's a need for an ini file

Reply:
The more modern method for project configuration is via a `pyproject.toml` so that could be an option if we still go with the config-file based approach but want to be "friendlier". I'm not sure what other approaches we should consider, if we still want to feature-flag this. I can imagine using an env variable, but I don't see how to make it sticky. Open to suggestions.

Reply:
After experimenting as well: From a user's standpoint, I agree that the current `.ini` file is a bit cumbersome. I think it is more convincing if you see this as more of an experimental feature right now, and you want to feature gate this by making users explicitly opt into this, and slowly phase it out as it gets more mature. 

On the other hand, I think a config file could be helpful for setting granular options for source map generation. Currently, the `compile()` method takes six or so optional arguments which complicates the interface. An idea we could consider is moving these to an `.ini` file so the developer could move those options out of the source code itself (and I think this would improve readability and improve backward compatibility if we wanted to change the inputs of the `compile()` method). 

Another tangential thought is that source maps should be verbose (rather than concise) by default - it feels like most other compilers follow this approach as well because compiling for production (with optimizations) and compiling for debugging are mutually exclusive use cases. I don't feel too strongly about this one, but just something that popped into my head.

Reply:
Or we could drop the `annotate_teal_*` parameters in favor of something like:

```python
@dataclass(frozen=True)
class SourceMapParams:
    ...
    annotation: bool = True
    annotation_columns: list[AnnotationColumn] = [
        AnnotationColumn.PYTEAL_PATH,
        AnnotationColumn.PYTEAL_LINE,
        AnnotationColumn.PYTEAL_CODE,
    ]
```

Reply:
I can definitely see keeping the `.ini` simple as possible argument. Regarding source map parameter building, I don't see this as a blocker to this current PR. We can also re-evaluate after getting community feedback as well.

Reply:
I'm planning to keep the `.ini` files while the sourcemapper is as slow as it is now. However, I'm keeping the discussion alive (in a sense) at #677 .

Resolving.